circus-train-gcp/src/main/java/com/hotels/bdp/circustrain/gcp/GCPCredentialCopier.java

```diff
@@ -35,57 +31,60 @@
 
 class GCPCredentialCopier {
   private static final Logger LOG = LoggerFactory.getLogger(GCPCredentialCopier.class);
-  private static final String WORKING_DIRECTORY = System.getProperty("user.dir");
-  private static final String RANDOM_STRING = UUID.randomUUID().toString() + System.currentTimeMillis();
-  private static final String CACHED_CREDENTIAL_NAME = "/ct-gcp-key-" + RANDOM_STRING + ".json";
-  private static final String HDFS_GS_CREDENTIAL_DIRECTORY = "hdfs:///tmp/ct-gcp-" + RANDOM_STRING;
-  private static final String HDFS_GS_CREDENTIAL_ABSOLUTE_PATH = HDFS_GS_CREDENTIAL_DIRECTORY + CACHED_CREDENTIAL_NAME;
+  static final String GCP_KEY_NAME = "ct-gcp-key.json";
```

Comment:
You probably still want this to have a random String on the end in case user configures a job to copy credentials to specific HDFS directory and it runs more than once. Consider moving the logic `dfsGCPCredentialAbsolutePath = new Path(dfsGCPCredentialDirectory, GCP_KEY_NAME)` into `DistributedFileSystemPathProvider` and append random String to the file there, return absolute path to credential file from `newPath` and use `path.getParent()` to get the `dfsGCPCredentialDirectory`

Reply:
Ah I see you add random string to a subdirectory in `DistributedFileSystemPathProvider`. Still move appending the key to the Path into that class so the logic is centralized